### ISSUE TITLE
DISCUSS: Upgrade GHC version (and LTS snapshot) used to build `stack`?

### ISSUE THREAD

**Issue by borsboom** [commercialhaskell/stack#4752]
TL;DR: I'm proposing we stick with GHC 8.2.2 for building `stack-2.0.*`.  Read on for the reasons why.

Up to now, we usually support building a new major `stack` release with the two most recent major GHC releases, and usually choose the most recent Stackage LTS snapshot as being the primary build environment (what we use to for integration tests and to build official binaries).  This would mean dropping support for GHC 8.2.2 and switching to GHC 8.6.4 as the primary build environment.

However, we are reconsidering this in light of the faster paced GHC release cycle, problems on the Windows platform, and reduced number of platforms supported with official bindists in more recent GHC versions.

Currently we're building `stack` with GHC 8.2.2 as the primary environment.  This GHC release has proved very stable, and has probably largest number of platforms bindists of any GHC release, including 32-bit Linux/Windows/FreeBSD, ARMv7 and AArch64, and CentOS 6.  GHC 8.4.4 drops ARM and CentOS 6.  GHC 8.6.4 further drops 32-bit Windows and FreeBSD (leaving only 32&64-bit Linux, 64-bit Windows, and macOS).  This all means that if we upgrade to a newer GHC, we'll have to stop building `stack` binaries for those platforms as well.  Note that lack of a bindist does not _necessarily_ mean that GHC doesn't support the platform, but as a rule we stick with official GHC bindists.

There have also been ongoing problems on Windows with GHC 8.4.* and GHC 8.6.* that make us hesitant to rely on them for `stack` releases (which must support Windows).

At this point, it doesn't seem like there's a compelling reason to upgrade to a newer GHC version as the primary environment, and some clear drawbacks, so I'm proposing we stick with the status-quo (same as stack-1.9.3) and don't upgrade GHC versions for next major release.  This means we'll support building `stack` with the most recent three major GHC versions.

Note: none of this has any effect on the versions of GHC you can use in projects that are built using Stack (we still support all the way back to GHC 7.8.4).  This only concerns building `stack` itself.

**Comment by snoyberg** [comment]
I agree with your analysis.

**Comment by snoyberg** [comment]
Decision made, closing
